onnx_tf/handlers/backend/pool_mixin.py

```diff
@@ -107,13 +100,58 @@ def dilated_pool():
     pooling_op = dilated_pool if pooling_type in ["MAX", "AVG", "LP"] else \
         dp.dilated_maxpool_with_argmax
 
-    # select the correct transpose ops depending on the input storage format
-    perm = get_perm_from_formats(compute_format, storage_format)
-
     def postprocess(pooled, argmax):
-      return (tf.transpose(pooled, perm=perm) if need_trans else pooled,
+
+      def convert_NHWC_indices_to_NCHW_indices(argmax):
+        # i - index in NCHW
+        # I - index in NHWC
+        # C - number of channels
+        # b - batch = I // CHW
+        # c - channel = I % C
+        # H - height
+        # W - weight
+        # I = i - c(HW - 1) + (C - 1)(i - bCHW - cHW)
+        # i = (I + c(HW - 1) + (C - 1)(bCHW + cHW))/C
+
+        # x_shape will always be in NCHW format here,
+        # because maxpool_with_argmax only support 2d input
+        x_shape = tf_shape(x)
+        N = x_shape[0]
+        C = x_shape[1]
+        H = x_shape[2]
+        W = x_shape[3]
+        HW = tf.math.multiply(H, W)
+        CHW = tf.math.multiply(C, HW)
+        argmax_b = tf.math.floordiv(argmax, CHW)
+        argmax_c = tf.math.floormod(argmax, C)
+        new_ind = tf.math.add(
+            argmax, tf.math.multiply(argmax_c, tf.math.subtract(HW, 1)))
+        new_ind = tf.math.add(
+            new_ind,
+            tf.math.multiply(
+                tf.math.subtract(C, 1),
+                tf.math.add(tf.math.multiply(argmax_b, CHW),
+                            tf.math.multiply(argmax_c, HW))))
+        new_ind = tf.math.floordiv(new_ind, C)
+
+        # add batch dimension into the argmax index
+        batch_offsets = tf.math.multiply(
+            tf.range(N, dtype=new_ind.dtype), CHW)
+        for _ in np.arange(len(new_ind.get_shape()) - 1):
```

Comment:
is there a reason to use np.arange instead of just python range for the loop? also maybe better way is to use new_ind.shape.rank or tf.rank(new_ind) to get the rank of the tensor. Tensor.get_shape() is deprecated.